### DISCUSSION TITLE
android11 can't use "IKEv2/IPSec MSCHAPv2" to connect strongswanVPN server

### DISCUSSION THREAD

**Discussion by GDkunban** [strongswan/strongswan#2192] [Q&A]
I use the android11 system as the client and use "IKEv2/IPSec MSCHAPv2" to connect  strongswanVPN server. When connecting, the client directly displays "failure", but the server is always connected, and then prompts
```
10[ENC] parsed IKE_AUTH request 1 [ IDi IDr SA TSi TSr CPRQ(ADDR ADDR6 DNS DNS6 MASK VER) ]
10[CFG] looking for peer configs matching 20.20.20.74[jp.6tv.eu]...119.123.56.93[jp.6tv.eu]
10[CFG] selected peer config 'IKEv2-EAP'
10[IKE] initiating EAP_IDENTITY method (id 0x00)
10[IKE] authentication of 'jp.6tv.eu' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
10[IKE] sending end entity cert "CN=jp.6tv.eu"
10[ENC] generating IKE_AUTH response 1 [IDr CERT AUTH EAP/REQ/ID]
10[ENC] splitting IKE message (1632 bytes) into 2 fragments
10[ENC] generating IKE_AUTH response 1 [EF(1/2)]
10[ENC] generating IKE_AUTH response 1 [EF(2/2)]
10[NET] sending packet: from 20.20.20.74[4500] to 119.123.56.93[17899] (1236 bytes)
10[NET] sending packet: from 20.20.20.74[4500] to 119.123.56.93[17899] (468 bytes)
13[IKE] sending keep alive to 119.123.56.93[17899]
12[JOB] deleting half open IKE_SA with 119.123.56.93 after timeout
```

**Comment by tobiasbrunner** [comment]
The client probably doesn't trust the server's certificate. Does the client have the CA certificate installed? Or is the server certificate self-signed?

**Reply by GDkunban** [reply]
Okay, I can connect using "IKEv2" of IOS16.1.2 system, but cannot connect using "IKEv2/IPSec MSCHAPv2" of Android 11 system.
How can I view more or more detailed error logs of the strongswanVPN server?

**Reply by GDkunban** [reply]
The server's certificate is not self-signed

**Reply by tobiasbrunner** [reply]
The server log won't help as it's a client problem. It should send the next IKE_AUTH request with an EAP-Identity response, but it doesn't. Again, probably because it doesn't trust the server certificate.

**Comment by TelDragon** [comment]
Hello, you can refer to it
https://github.com/strongswan/strongswan/discussions/1843
